### PR TITLE
feat(model): add DM0 support

### PR DESCRIPTION
**Pull request by xb534**
### Description
Add support for DM0 model from DEXMAL in the embodied policy training framework, enabling PPO training with the DM0 policy.

### Motivation and Context
DM0 is a new diffusion-based policy model from DEXMAL. This PR integrates DM0 into the RLinf framework so it can be trained with PPO, similar to the existing Pi0 support.

### How has this been tested?
Tested PPO training with DM0 on the LIBERO-Spatial benchmark using the `libero_spatial_ppo_dexbotic_dm0` config.

### Additional information (optional, e.g., figures and logs):
N/A

### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (Document-only update)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: examples/embodiment/config/model/dexbotic_pi.yaml [hunk 1]
```diff
@@ -18,13 +18,6 @@ use_proprio: True
 num_steps: 10  # Number of diffusion/denoising steps
 add_value_head: True  # Add value head for PPO
 
-# RL training specific settings
-safe_get_logprob: False  # Use simplified logprob calculation (False = full Gaussian)
```
Comment (WinstonWmj) [inline examples/embodiment/config/model/dexbotic_pi.yaml:22]: Maybe you should not remove these parameters. Because they will be used in rlinf/models/embodiment/dexbotic_dm0/__init__.py

File: examples/embodiment/config/libero_spatial_ppo_dexbotic_dm0.yaml [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  micro_batch_size: 32
+  global_batch_size: 8192
+  seed: 42
+  enable_offload: False
+
+  model:
+    model_path: "/path/to/model/libero-db-dm0" # https://huggingface.co/Dexmal/DM0-libero
+    precision: "bf16"
+    num_action_chunks: 5
+    num_steps: 3
```
Comment (WinstonWmj) [inline examples/embodiment/config/libero_spatial_ppo_dexbotic_dm0.yaml]: db_pi0 and db_dm0 use different `num_steps`?

File: rlinf/models/embodiment/dexbotic_dm0/dm0_policy.py [hunk 3]
```diff
@@ -0,0 +1,777 @@
...
+        entropy = 0.5 * torch.log(2 * math.pi * math.e * (sigma_safe**2))
+        return entropy
+
+    @torch.no_grad()
+    def sample_actions(
+        self, processed_obs, noise=None, mode="train", compute_values=True
+    ):
+        original_training_mode = self.training
+        self.eval()
+        try:
```
Comment (WinstonWmj) [inline rlinf/models/embodiment/dexbotic_dm0/dm0_policy.py]: Could you implement this in the same way as pi0 (rlinf/models/embodiment/openpi/openpi_action_model.py)? I’m not sure whether this (try and finally to make sure the train/eval mode is correct) is strictly necessary.

File: docs/source-en/rst_source/examples/embodied/dexbotic.rst [hunk 4]
```diff
@@ -115,7 +115,9 @@ Install dependencies directly in your environment:
 Model Download
 --------------
 
-Before starting training, download the Dexbotic SFT model from HuggingFace:
+**π\ :sub:`0`\ model**
```
Comment (WinstonWmj) [inline docs/source-en/rst_source/examples/embodied/dexbotic.rst]: There is an RST syntax issue in "Dexbotic (π:sub:0.5-based VLA)", which causes incorrect rendering.

File: docs/source-zh/rst_source/examples/embodied/dexbotic.rst [hunk 5]
```diff
@@ -105,7 +105,9 @@ Dexbotic 直接使用环境提供的自然语言任务描述作为语言模型
 模型下载
 --------------
 
-开始训练前，请从 HuggingFace 下载 Dexbotic SFT 模型：
+**π\ :sub:`0`\ 模型**
```
Comment (WinstonWmj) [inline docs/source-zh/rst_source/examples/embodied/dexbotic.rst]: There is an RST syntax issue in "Dexbotic (π:sub:0.5-based VLA)", which causes incorrect rendering.

**Review by WinstonWmj — CHANGES_REQUESTED** [review]
@xb534 hi there, i have some comments you need to solve.

File: docs/source-zh/rst_source/examples/embodied/dexbotic.rst (hunk 5 above)
Comment (xb534) [reply docs/source-zh/rst_source/examples/embodied/dexbotic.rst]: Thanks for catching this — it's already fixed in the latest commit.

File: examples/embodiment/config/libero_spatial_ppo_dexbotic_dm0.yaml (hunk 2 above)
Comment (xb534) [reply examples/embodiment/config/libero_spatial_ppo_dexbotic_dm0.yaml]: It should be 4 as well — will fix it.

**Comment by Iron-Wph** [comment]
@xb534 
Hello, I tested the training of DM0 using the following command, and the results are shown below.
```
bash examples/embodiment/run_embodiment.sh libero_spatial_ppo_dexbotic_dm0
```
PixPin_2026-04-27_22-59-52

I also tested the training of PI0 using the following command, and encountered the following error.
```
bash examples/embodiment/run_embodiment.sh libero_spatial_ppo_dexbotic_pi0
```
[run_embodiment.log](https://github.com/user-attachments/files/27131077/run_embodiment.log)


Additionally, I ran the evaluation of DM0 using the following command, and encountered the following error.
```
   python toolkits/eval_scripts_dexbotic/libero_eval.py \
      --config_name dm0_libero \
      --pretrained_path /workspace/dataset/DM0-libero \
      --task_suite_name libero_spatial \
      --num_trials_per_task 50 \
      --action_chunk 10 \
      --num_steps 10
```
[dm0_eval.log](https://github.com/user-attachments/files/27131258/dm0_eval.log)

File: examples/embodiment/config/model/dexbotic_pi.yaml (hunk 1 above)
Comment (xb534) [reply examples/embodiment/config/model/dexbotic_pi.yaml:22]: Already fixed in the latest commit.

File: rlinf/models/embodiment/dexbotic_dm0/dm0_policy.py (hunk 3 above)
Comment (xb534) [reply rlinf/models/embodiment/dexbotic_dm0/dm0_policy.py]: Already fixed in the latest commit.

**Comment by xb534** [comment]
@Iron-Wph  Thanks for the review! I've addressed the issues you pointed out — please take another look when you have a chance.

**Comment by Iron-Wph** [comment]
@xb534 
Hello, I have re-tested the PR and the results are as expected. 

1. Below are the outputs obtained from running the commands.
```
bash examples/embodiment/run_embodiment.sh libero_spatial_ppo_dexbotic_pi0
```
PixPin_2026-04-29_14-28-25

2. Below are the outputs obtained from running the commands.
```
   python toolkits/eval_scripts_dexbotic/libero_eval.py \
      --config_name dm0_libero \
      --pretrained_path /path/to/checkpoint \
      --task_suite_name libero_spatial \
      --num_trials_per_task 50 \
      --action_chunk 10 \
      --num_steps 10
```
PixPin_2026-04-29_14-33-52

3. Below are the outputs obtained from running the commands.
```
   python toolkits/eval_scripts_dexbotic/libero_eval.py \
      --config_name db_pi0_libero \
      --pretrained_path /path/to/checkpoint \
      --task_suite_name libero_spatial \
      --num_trials_per_task 50 \
      --action_chunk 5 \
      --num_steps 10
```
PixPin_2026-04-29_14-34-02

**Review by guozhen1997 — APPROVED** [review]

**Review by andylin-hao — APPROVED** [review]
